src/core/infer_request.h

```diff
@@ -388,6 +390,35 @@ class InferenceRequest {
   // contain only the minimum content required for a null request.
   static InferenceRequest* CopyAsNull(const InferenceRequest& from);
 
+  void CaptureRequestStartNs()
+  {
+    struct timespec ts;
+    clock_gettime(CLOCK_MONOTONIC, &ts);
+    request_start_ns_ = TIMESPEC_TO_NANOS(ts);
+  }
+
+  void CaptureQueueStartNs()
+  {
+    struct timespec ts;
+    clock_gettime(CLOCK_MONOTONIC, &ts);
+    queue_start_ns_ = TIMESPEC_TO_NANOS(ts);
+  }
+
+  uint64_t QueueStartNs() { return queue_start_ns_; }
+
+  // Report the statistics to stats collectors associated with the request.
+  // Duration and timestamps provide two granularities for stats collectors.
+  void Report(
+      bool success, int device, uint64_t compute_start_ns,
+      uint64_t compute_input_end_ns, uint64_t compute_output_start_ns,
+      uint64_t compute_end_ns, uint64_t compute_input_duration_ns,
+      uint64_t compute_infer_duration_ns, uint64_t compute_output_duration_ns);
+
+  // Associate a stats collector with the request. We may generalize it
+  // and expose it for user to use.
+  void AddStatsCollector(
```

Comment:
SetSecondaryStatsAggregator()
The comment should be something like "Statistics for each request are aggregated into the corresponding backend's statistics. Optionally this function may be used to add an additional aggregator where statistics are also aggregated"